deps/xdiff/CMakeLists.txt

```diff
@@ -1,14 +1,29 @@
 
-file(GLOB SRC_XDIFF "*.c" "*.h")
-list(SORT SRC_XDIFF)
-
-add_library(xdiff OBJECT ${SRC_XDIFF})
+add_library(xdiff STATIC
```

Comment:
Why is this preferable to just using a glob?

Reply:
CMake recommands to avoid GLOB especially with source files : 

https://cmake.org/cmake/help/latest/command/file.html#glob
``
Note

We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. The CONFIGURE_DEPENDS flag may not work reliably on all generators, or if a new generator is added in the future that cannot support it, projects using it will be stuck. Even if CONFIGURE_DEPENDS works reliably, there is still a cost to perform the check on every rebuild. 
``

I made the changes in some places when I had to modify a lot the CMake logic. However it should be possiblle to supress all the GLOBs. This is quite easy and could be finish in an other dedicated PR.